src/routes/redux/detailsSlice.js

```diff
@@ -0,0 +1,68 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+export const detailsSlice = createSlice({
+  name: "details",
+  initialState: {
+    value: {
+      name: "",
+      lastName: "",
+      phone: "",
+      address: "",
+      shirtSize: "",
+      pantsSize: "",
+      budget: "",
+      neverWear: [],
+      alwaysWear: [],
+    },
+  },
+  reducers: {
+    updateDetails: (state, action) => {
+      state.value = {
+        ...state.value,
+        [action.payload.fieldName]: action.payload.value,
```

Comment:
you can achieve the same with a simpler solution
what you do here is spread the state.value and add an additional field
you can just do 
`state.value[action.payload.fieldName]: action.payload.value`